modules/users/tests/server/user-block.server.routes.tests.js

```diff
@@ -0,0 +1,279 @@
+const should = require('should');
+const request = require('supertest');
+const path = require('path');
+const mongoose = require('mongoose');
+const User = mongoose.model('User');
+const express = require(path.resolve('./config/lib/express'));
+
+/**
+ * Globals
+ */
+let app;
+let agent;
+
+/* users of the test */
+let alice;
+let bob;
+let carol;
+
+/**
+ * User routes tests
+ */
+describe('User block - user', function () {
+  before(function (done) {
+    // Get application
+    app = express.init(mongoose.connection);
+    agent = request.agent(app);
+
+    done();
+  });
+
+  beforeEach(function (done) {
+    // Alice user
+    const aliceCredentials = {
+      username: 'alice_the_blocker',
+      password: 'TR-I$Aw3$0m4',
+    };
+
+    const aliceProfile = {
+      public: true,
+      firstName: 'Alice',
+      lastName: 'Doe',
+      displayName: 'Alice Doe',
+      email: 'alice@example.org',
+      emailToken: 'initial email token',
+      username: aliceCredentials.username.toLowerCase(),
+      password: aliceCredentials.password,
+      provider: 'local',
+    };
+    const aliceUser = new User(aliceProfile);
+    alice = {
+      credentials: aliceCredentials,
+      user: aliceUser,
+      profile: aliceProfile,
+    };
+
+    // Bob user
+    const bobCredentials = {
+      username: 'bob_the_blocked',
+      password: 'TR-I$Aw3$0m4',
+    };
+    const bobProfile = {
+      public: true,
+      firstName: 'Bob',
+      lastName: 'Doe',
+      displayName: 'bob doe',
+      email: 'bob@example.org',
+      emailToken: 'initial email token',
+      username: bobCredentials.username.toLowerCase(),
+      password: bobCredentials.password,
+      provider: 'local',
+    };
+    const bobUser = new User(bobProfile);
+    bob = { credentials: bobCredentials, user: bobUser, profile: bobProfile };
+
+    // Carol user
+    const carolCredentials = {
+      username: 'carol_the_blocker',
+      password: 'TR-I$Aw3$0m4',
+    };
+    const carolProfile = {
+      public: true,
+      firstName: 'Carol',
+      lastName: 'Doe',
+      displayName: 'Carol Doe',
+      email: 'carol@example.org',
+      emailToken: 'initial email token',
+      username: carolCredentials.username.toLowerCase(),
+      password: carolCredentials.password,
+      provider: 'local',
+    };
+    const carolUser = new User(carolProfile);
+    carol = {
+      credentials: carolCredentials,
+      user: carolUser,
+      profile: carolProfile,
+    };
+
+    // Save alice and bob to test db
+    Promise.all([aliceUser.save(), bobUser.save(), carolUser.save()])
```

Comment:
I know this is how most of the app does this, but it can also be simplified nowadays to this:

https://github.com/Trustroots/trustroots/blob/aaa8b9ed0269c8e4ea157bcc6097db18b7eb5719/modules/admin/tests/server/admin.audit-log.server.routes.tests.js#L12-L19

Not a blocker for merging.

Reply:
Nice! Much more simple.